### PR TITLE
Disable file logging & simplfy log filtering

### LINKED ISSUE #23102: Disable File logging and simplify setting log levels
Logging to file causes a number of problems on systems, particularly if multiple instances of Mantid are started.

e.g.
- on windows the second instance will not log at all as the log file is locked
- on linux all instances log to the same file making it hard to follow
- also shared network home directories can lead to spikes in network traffic for things that log a lot.
- in 10 years of Mantid no current developer can remember the log files actually being helpful in there current state.

Also the current approach of multiple log levels is not justifiable once file logging is removed.  Simplify it so that all logs have the same priority level.

### PR DESCRIPTION
**Pull request by NickDraper**
**Description of work.**
Removed file logging, and changed log filtering so you cannot set separate log filters for each log (few people understood this anyway).

**To test:**
1. Check unit and system tests work
1. Start mantiplot, play with the log levels in the message display, check they still work.
1. Check that the message level is remembered when closing and restarting Mantid.

Fixes #23102 

Does this update require release notes?
- [x] Yes
- [ ] No

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by keeeto — APPROVED** [review]
Tested on OSX.

Technically works as described and didn't seem to interfere with any other functionality when I was playing around with Mantid. The .log file is no longer created.

I suggest someone else needs to do the code review!!

File: Framework/Kernel/src/ConfigService.cpp [hunk 1]
```diff
@@ -377,10 +368,8 @@ void ConfigServiceImpl::loadConfig(const std::string &filename,
     // there was a problem loading the file - it probably is not there
     std::cerr << "Problem loading the configuration file " << filename << " "
               << e.what() << '\n';
-    if (!append) {
-      // if we have no property values then take the default
-      m_PropertyString = defaultConfig();
-    }
+    std::cerr << "Mantid is unable to start.\n" << std::endl;
+    throw;
```
Comment (martyngigg) [inline Framework/Kernel/src/ConfigService.cpp]: Agreed that we just stop now :+1:

File: Framework/Properties/Mantid.properties.template [hunk 2]
```diff
@@ -169,16 +169,11 @@ graph1d.autodistribution = On
 
 # logging configuration
 # root level message filter (This sets a minimal level possible for any channel)
-logging.loggers.root.level = information
+logging.loggers.root.level = notice
```
Comment (martyngigg) [inline Framework/Properties/Mantid.properties.template]: Agreed that this is now a sensible default.

**Review by martyngigg — APPROVED** [review]
The code changes look good to me.
